cpp/src/Ice/Incoming.cpp

```diff
@@ -1,672 +0,0 @@
-        }
-        handleException(exc);
-    }
-    catch (const LocalException&)
-    {
-        _responseHandler->invokeException(_current.requestId, current_exception(), 1); // Fatal invocation exception
-    }
-}
-
-void
-Incoming::response()
-{
-    writeEmptyParams();
-    completed();
-}
-
-void
-Incoming::response(MarshaledResult&& result)
-{
-    setMarshaledResult(std::move(result));
-    completed();
-}
-
-void
-Incoming::warning(const Exception& ex) const
-{
-    Warning out(_os.instance()->initializationData().logger);
-
-    ToStringMode toStringMode = _os.instance()->toStringMode();
-
-    out << "dispatch exception: " << ex;
-    out << "\nidentity: " << identityToString(_current.id, toStringMode);
-    out << "\nfacet: ";
-    out << escapeString(_current.facet, "", toStringMode);
-    out << "\noperation: " << _current.operation;
-
-    if (_current.con)
-    {
-        try
-        {
-            for (Ice::ConnectionInfoPtr connInfo = _current.con->getInfo(); connInfo; connInfo = connInfo->underlying)
-            {
-                Ice::IPConnectionInfoPtr ipConnInfo = dynamic_pointer_cast<Ice::IPConnectionInfo>(connInfo);
-                if (ipConnInfo)
-                {
-                    out << "\nremote host: " << ipConnInfo->remoteAddress << " remote port: " << ipConnInfo->remotePort;
-                    break;
-                }
-            }
-        }
-        catch (const Ice::LocalException&)
-        {
-            // Ignore.
-        }
-    }
-}
-
-void
-Incoming::warning(std::exception_ptr ex) const
-{
-    Warning out(_os.instance()->initializationData().logger);
-    ToStringMode toStringMode = _os.instance()->toStringMode();
-
-    out << "dispatch exception: ";
-
-    try
-    {
-        rethrow_exception(ex);
-    }
-    catch (const std::exception& e)
-    {
-        out << "std::exception: " << e.what();
-    }
-    catch (...)
-    {
-        out << "unknown c++ exception";
-    }
-    out << "\nidentity: " << identityToString(_current.id, toStringMode);
-    out << "\nfacet: " << escapeString(_current.facet, "", toStringMode);
-    out << "\noperation: " << _current.operation;
-
-    if (_current.con)
-    {
-        for (Ice::ConnectionInfoPtr connInfo = _current.con->getInfo(); connInfo; connInfo = connInfo->underlying)
-        {
-            Ice::IPConnectionInfoPtr ipConnInfo = dynamic_pointer_cast<Ice::IPConnectionInfo>(connInfo);
-            if (ipConnInfo)
-            {
-                out << "\nremote host: " << ipConnInfo->remoteAddress << " remote port: " << ipConnInfo->remotePort;
-                break;
-            }
-        }
-    }
-}
-
-bool
-Incoming::servantLocatorFinished()
-{
-    assert(_locator && _servant);
-    try
-    {
-        _locator->finished(_current, _servant, _cookie);
-        return true;
-    }
-    catch (...)
-    {
-        handleException(current_exception());
-    }
-    return false;
-}
-
-void
-Incoming::handleException(std::exception_ptr exc)
```

Comment:
This largely moved to OutgoingResponse.cpp - except for the logging code (=> LoggerMiddleware) and the observer code (=> ObserverMiddleware).